ludwig/models/modules/convolutional_modules.py

```diff
@@ -544,60 +544,44 @@ def __init__(
         for i in range(num_layers):
             layer = self.layers[i]
-            with tf.variable_scope('conv_{}'.format(i)):
-                # Convolution Layer
-                filter_shape = [
-                    layer['filter_size'],
-                    layer['filter_size'],
-                    prev_num_filters,
-                    layer['num_filters']
-                ]
-                layer_output = conv_2d_layer(hidden, filter_shape,
-                                             [layer['num_filters']],
-                                             stride=layer['stride'],
-                                             padding='SAME',
-                                             activation=layer['activation'],
-                                             norm=layer['norm'],
-                                             dropout=layer['dropout'],
-                                             dropout_rate=dropout_rate,
-                                             regularizer=regularizer if layer[
-                                                 'regularize'] else None,
-                                             initializer=layer['initializer'],
-                                             is_training=is_training)
-                prev_num_filters = layer['num_filters']
-                logger.debug('  conv_{}: {}'.format(i, layer_output))
 
-                # Pooling
-                if layer['pool_size'] is not None:
-                    pool_size = layer['pool_size']
-                    pool_kernel_size = [1, pool_size, pool_size, 1]
-                    pool_stride = layer['pool_stride']
-                    if pool_stride is not None:
-                        pool_kernel_stride = [1, pool_size, pool_size, 1]
-                    else:
-                        pool_kernel_stride = [1, pool_stride, pool_stride, 1]
-                    layer_output = tf.nn.max_pool(
-                        layer_output,
-                        ksize=pool_kernel_size,
-                        strides=pool_kernel_stride,
-                        padding='SAME',
-                        name='pool_{}'.format(i)
-                    )
+            # TODO add filter size tuples
+            # TODO provide regularization penalty
+            # TODO provide bias and activity regularizers
+            # TODO add initializers
 
-                    logger.debug('  pool_{}: {}'.format(
-                        i, layer_output))
-
-            hidden = layer_output
-            logger.debug('  hidden: {0}'.format(hidden))
+            conv_layer = tf.keras.layers.Conv2D(
+                layer['num_filters'],
+                layer['filter_size'],
+                kernel_regularizer=regularizer if layer['regularize'] else None,
+            )
+            hidden = conv_layer(hidden)
+
+            if layer['dropout'] and dropout_rate > 0:
+                hidden = tf.nn.dropout(hidden, rate=dropout_rate)
```

Comment:
I think there's a tf2 dropout layer too https://www.tensorflow.org/api_docs/python/tf/keras/layers/Dropout
Also consider dropout will soon be unified in one singe number, not rate and boolean as of now., so there will be no if

Reply:
Yep, updated.